addons/cloud_storage/models/mail_mail.py

```diff
@@ -0,0 +1,49 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import re
+
+from odoo import models, tools
+
+
+class MailMail(models.Model):
+    """Add the mass mailing campaign data to mail"""
+    _inherit = ['mail.mail']
+
+    def _prepare_outgoing_list(self, mail_server=False, recipients_follower_status=None):
+        outgoing_list = super()._prepare_outgoing_list(
+            mail_server=mail_server,
+            recipients_follower_status=recipients_follower_status
+        )
+
+        # Turn remaining attachments into links appended to an email if they are cloud attachments
+        for mail, outgoing in zip(self, outgoing_list):
```

Comment:
This logic is not correct
you can find `self.ensure_one()` in the `super()._prepare_outgoing_list(`

`outgoing_list` are emails to different people for the same mail record. This line only handles the first `outgoing`.

BTW. The `outgoing['attachments']` are shared list, you can replace its data it in place for the first one.` But `outgoing['body']` are personalized.